### PR TITLE
Add `indexerLedgerForEval` interface

### PR DESCRIPTION
**Pull request by tolikzinovyev**
## Summary

Add a new ledger interface for Indexer that is significantly simpler and allows implementing batching in a more straight-forward way. As a result, the Indexer code doesn't need to know the specifics of go-algorand and doesn't need to implement its own accounts cache for batching.

### CODE REVIEW

File: ledger/eval.go [hunk 1]
```diff
@@ -1525,8 +1525,108 @@ func (vb ValidatedBlock) WithSeed(s committee.Seed) ValidatedBlock {
...
+	Address basics.Address
+	Exists  bool
+}
+
+// A ledger interface that Indexer implements. This is a simplified version of the
+// ledgerForEvaluator interface. Certain functions that the evaluator doesn't use
+// in the trusting mode are excluded, and the present functions only request data
+// at the latest round.
+type indexerLedgerForEval interface {
+	BlockHdr(basics.Round) (bookkeeping.BlockHeader, error)
```
Comment (tsachiherman) [inline ledger/eval.go:1539]: when do you need different block header rounds ?

Comment (tolikzinovyev) [reply ledger/eval.go:1539]: We only need to previous block header. Do you think I should simplify it?

Comment (tsachiherman) [reply ledger/eval.go:1539]: Yes; Naming it LatestBlockHdr() would be a good name... and in `indexerLedgerConnector. BlockHdr` you should test that round == blockHdr.Round(), or error. ( i.e. verify that the ledger wasn't really referring to a header which isn't the latest )

Comment (tsachiherman) [reply ledger/eval.go:1539]: btw - same applied for Totals()

**Review by tsachiherman — COMMENTED** [review]
Do you think that we can move the indexer related implementation to its own file ?
maybe indexereval.X ?

File: ledger/eval.go (hunk 1 above)
Comment (tolikzinovyev) [reply ledger/eval.go]: done

**Comment by tolikzinovyev** [comment]
> Do you think that we can move the indexer related implementation to its own file ?
> maybe indexereval.X ?

Good idea, I moved the indexer code.

**Review by tsachiherman — DISMISSED** [review]

**Comment by tsachiherman** [comment]
Looks good - could you please verify it passes all the auto tests ?

**Review by tsachiherman — APPROVED** [review]

**Pull request merged**